src/sidebar/components/user-menu.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+
+const bridgeEvents = require('../../shared/bridge-events');
+const { isThirdPartyUser } = require('../util/account-id');
+const serviceConfig = require('../service-config');
+const { withServices } = require('../util/service-context');
+
+const Menu = require('./menu');
+const MenuSection = require('./menu-section');
+const MenuItem = require('./menu-item');
+
+/**
+ * A menu with user and account links.
+ *
+ * This menu will contain different items depending on service configuration,
+ * context and whether the user is first- or third-party.
+ */
+function UserMenu({ auth, bridge, inSidebar, onLogout, serviceUrl, settings }) {
+  const isThirdParty = isThirdPartyUser(auth.userid, settings.authDomain);
+  const service = serviceConfig(settings);
+
+  // If there is service configuration set, check for whether the given
+  // `feature` is supported. If no service configuration is present,
+  // assume support for the feature.
+  const serviceSupports = feature => !service || !!service[feature];
```

Comment:
I was careful to copy this logic over from the old component but I want to be very clear here: the logic is such that if a `service` object is not present, the feature in question is considered "supported." Is this actually correct?